packages/frontend-api/src/Component/Price/MoneyFormatterHelper.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Component\Price;
+
+use CommerceGuys\Intl\Currency\CurrencyRepositoryInterface;
+use Shopsys\FrameworkBundle\Component\CurrencyFormatter\CurrencyFormatterFactory;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Model\Pricing\Currency\CurrencyFacade;
+
+class MoneyFormatterHelper
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Pricing\Currency\CurrencyFacade
+     */
+    protected $currencyFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\CurrencyFormatter\CurrencyFormatterFactory
+     */
+    protected $currencyFormatterFactory;
+
+    /**
+     * @var \CommerceGuys\Intl\Currency\CurrencyRepositoryInterface
+     */
+    protected $intlCurrencyRepository;
+
+    /**
+     * MoneyFormatterHelper constructor.
+     *
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Currency\CurrencyFacade $currencyFacade
+     * @param \Shopsys\FrameworkBundle\Component\CurrencyFormatter\CurrencyFormatterFactory $currencyFormatterFactory
+     * @param \CommerceGuys\Intl\Currency\CurrencyRepositoryInterface $intlCurrencyRepository
+     */
+    public function __construct(
+        CurrencyFacade $currencyFacade,
+        CurrencyFormatterFactory $currencyFormatterFactory,
+        CurrencyRepositoryInterface $intlCurrencyRepository
+    ) {
+        $this->currencyFacade = $currencyFacade;
+        $this->currencyFormatterFactory = $currencyFormatterFactory;
+        $this->intlCurrencyRepository = $intlCurrencyRepository;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $money
+     * @param int $domainId
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    public function getMoneyFormattedAsDecimalWithFractionDigitsByCurrencySetting(Money $money, int $domainId): Money
+    {
+        $locale = 'en';
+        $currency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domainId);
+
+        $currencyFormatter = $this->currencyFormatterFactory->createByLocaleAndCurrency($locale, $currency);
+        $intlCurrency = $this->intlCurrencyRepository->get(
+            $currency->getCode(),
+            $locale
+        );
+
+        return Money::create(
+            $currencyFormatter->format(
+                $money->getAmount(),
+                $intlCurrency->getCurrencyCode(),
+                [
+                    'currency_display' => 'none',
+                    'use_grouping' => false,
+                ]
+            )
+        );
```

Comment:
I do not understand this at all... Money should not include any formatting, it is supposed to be value only. My head is spinning from thinking, if and how this could work.

Reply:
If you need formatting on output, I think better place to do it is `MoneyType::serialize` method, as it responsible for outputing all money values in API.

Reply:
> I do not understand this at all... Money should not include any formatting, it is supposed to be value only. My head is spinning from thinking, if and how this could work.

This is the only code responsible for proper price formatting. I have already discussed it with @grossmannmartin about not ideal flow of doing that. Sadly it is what we have got.

In future I would like to make the formatting more readable/extandable/clean.

> If you need formatting on output, I think better place to do it is MoneyType::serialize method, as it responsible for outputing all money values in API.

This is really good point! Thanks!

Reply:
Sadly it cannot be done as the MoneType uses static methods. Because of that we are not able to format it there.

Reply:
In such case some documentation describing, how people should format Money type in FE API would be appreciated.